### DISCUSSION TITLE
Dynamically linking lnd

### DISCUSSION THREAD

**Discussion by cndolo** [lightningnetwork/lnd#8878] [Q&A]
Hi,

I am trying to compile a dynamically linked `lnd` executable but have had no success so far using the following steps:

1. clone the lnd repo
2.` go install -buildmode=shared -linkshared  std`
3.  Changed [l. 34 of the Makefile](https://github.com/lightningnetwork/lnd/blob/v0.18.1-beta/Makefile#L34) to `GOBUILD := $(LOOPVARFIX) go build -v -linkshared]`

Running `make build` fails with the following error:

```
link: duplicated definition of symbol github.com/lightningnetwork/lnd/fn.Option[go.shape.struct { github.com/lightningnetwork/lnd/tlv.recordType *github.com/lightningnetwork/lnd/tlv.tlvType0; Val *github.com/decred/dcrd/dcrec/secp256k1/v4.PublicKey }].UnwrapOrFunc.func1, from github.com/lightningnetwork/lnd and main
make: *** [Makefile:107: build] Error 1
```

Also tried prepending `GCO_ENABLED=1` to [l. 107/108](https://github.com/lightningnetwork/lnd/blob/v0.18.1-beta/Makefile#L107) but the error persists.

Would anyone happen to have any ideas of what (else) I need to change in the Makefile in order to produce a dynamically linked executable?

Thanks.

**Comment by ffranr** [comment]
> A "dynamically linked executable" is a program that relies on shared libraries (or dynamic link libraries, DLLs) that are not embedded within the executable file itself. Instead, these libraries are loaded into memory at runtime by the operating system, which allows multiple programs to share the same library code, reducing redundancy and saving memory.

^ this is what I've found online.

Can we take a step back please and can you help me understand what you're trying to achieve? I don't think LND relies on many shared libraries as it currently stands.
